src/backend/web/validation.js

```diff
@@ -50,4 +51,41 @@ function validateNewFeed() {
   };
 }
 
-module.exports.validateNewFeed = validateNewFeed;
+// query validation starts here
+// queryValidation rules
+const queryValidationRules = () => {
+  return [
+    // text must be between 1 and 256
+    check('text')
+      .exists()
+      .withMessage('text should not be empty')
+      .isLength({ max: 256, min: 1 })
+      .withMessage('text should be between 0 to 256 characters'),
+    // filter must exists
+    check('filter')
+      .exists()
+      .withMessage('filter should exist')
+      .contains('post' || 'author')
+      .withMessage('invalid filter value'),
+  ];
+};
+
+// queryValidationFunction
+const validateQuery = (req, res, next) => {
+  const errors = validationResult(req);
+  if (errors.isEmpty()) {
+    return next();
+  }
+  const extractedErrors = [];
+  errors.array().map((err) => extractedErrors.push({ [err.param]: err.msg }));
```

Comment:
`map()` returns an array, so you don't need to `push()` manually:

```js
const extracted Errors = errors.array().map((err) => ({ [err.param]: err.msg }));
```

Reply:
Incorporated in [7dd761b](https://github.com/Seneca-CDOT/telescope/pull/1334/commits/7dd761bc7b9281cd31a572a30424e0c41235cb6b)